src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs

```diff
@@ -1890,6 +1892,28 @@ public void Dispose()
             this.orchestrationSessionManager.Dispose();
         }
 
+        /// <summary>
+        /// Gets the status of all orchestration instances with paging that match the specified conditions.
+        /// </summary>
+        public Task GetOrchestrationStateWithFiltersAsync(OrchestrationStatusQueryCondition condition, CancellationToken cancellationToken)
+        {
+            var transferredCondition = TransferToAzureStorageCondition(condition);
+            return GetOrchestrationStateAsync(transferredCondition, condition.PageSize, condition.ContinuationToken, cancellationToken);
+        }
+
+        internal OrchestrationInstanceStatusQueryCondition TransferToAzureStorageCondition(OrchestrationStatusQueryCondition condition)
```

Comment:
Let's simplify the name a bit. Also, is it okay for this to be `private` (default visibility) instead of `internal`? 
```suggestion
        OrchestrationInstanceStatusQueryCondition ToAzureStorageCondition(OrchestrationStatusQueryCondition condition)
```